### PR TITLE
Ss questions by level

### LINKED ISSUE #41: [BE] User Story: Questions by level
As a user, I should be able to get speech and grammar questions based on their proficiency level

### PR DESCRIPTION
**Pull request by sschipke**
## What's this PR do?
1. Add a route for getting speech questions by level
1. Add a route for getting grammar questions by level

### Where should the reviewer start?

- [ ] app.py

### How should this be manually tested?
- `GET` request to `/speech_questions/:level`
- `GET` request to `/grammar_questions/:level`

### Releveant tickets
closes #41

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,13 +11,6 @@ before_script:
 script:
 - python3 manage.py test
 deploy:
-  provider: heroku
-  api_key:
-    secure: TmigMa2OGKyvftBtg9g2OpKdi83Ebg8O0i+v32nktt3/HIBD9d55BxtFYA6S4pRaFk6DtV/iIWtMLPn97DEEneIkYiNeTiDPoeuVnyhXqxJ2I4iin33vg9hDVArQxDY0BK9/vCBNYb3YR7HQjGRdNcx55XGsGlRyBwqSIWZv8W6NnPqozejQJjuSWsM3yb9a9x0WMVluukJEsR2oBWgXKm5LJqN4ofar4T2W5WEyaNZ7smZFOLZCKz9uVgrn7n2S8BMP0lBnPCgQeZK0vDAz08SBcci8550S6Nc+hW8D1DLK0qIpdky/4RAj+xAzNcpjAHTMKjDlZZPgC/F5LxFpLNFctWbeAL/QzAHa0wdKpzntScDaTlpYNMbphWBuxmW8/4VZA9SlnD+XLzj8XYPc/4ixBJLl5jc2Gemxs3mTRd6XMhYbZWlLI1XtdE+emfmY3kGtnStFRTZnEUzwRrSqvJ+PKk3I4QBlLh8tpcPuYTSqhtKlVR/vUxm2nN9ZXGI755GapqAjftTIjdpUlXMwa/juL4q2uMx6AmrQMXrLcjSDKttUb+Mafd7peDgTqfLt8kmVeF9EP3FoTIO7R5tNTi4FOVJegU0TJT0SCP3H98tFIcI8tV/x2LnEd2vIdj3MuVKs6iyUnA8xzc6bGOxo2G/H5G5WNaMdRTrOv/aJmik=
-  app: language-learner-be
-  on:
-    repo: nancylee713/AILanguageTutor_BE
-    branch: master
```
Comment (sschipke) [inline .travis.yml]: We are no longer deploying this, correct? We'll just go from staging?

File: .travis.yml [hunk 2]
```diff
@@ -11,13 +11,6 @@ before_script:
 script:
 - python3 manage.py test
 deploy:
```
Comment (nancylee713) [inline .travis.yml]: this line also needs to be deleted

File: app.py [hunk 3]
```diff
@@ -2,9 +2,6 @@
 from flask import Flask, request, jsonify, json, session
```
Comment (nancylee713) [inline app.py]: we could also delete `request`, `json`, and `session` since they're not being used in this file

File: app.py [hunk 4]
```diff
@@ -87,6 +82,23 @@ def get_speech_questions():
...
+def get_speech_questions_by_level(level):
+    possible_levels = ['beginner', 'intermediate', 'advanced']
+    if level in possible_levels:
+        try:
+            questions = SpeechQuestion.query.filter_by(level=level).all()
+            return jsonify([e.serialize() for e in questions])
+        except Exception as e:
+            return(str(e))
+    else:
+        return jsonify({"error": "Level must be either beginner, intermediate, or advanced NOT -{}-".format(level)}), 404
```
Comment (nancylee713) [inline app.py]: Nice error handling!

File: .travis.yml (hunk 1 above)
Comment (nancylee713) [reply .travis.yml]: correct

File: .travis.yml (hunk 2 above)
Comment (sschipke) [reply .travis.yml]: Just the deploy line or all of 
```
script:
- python3 manage.py test
```

File: app.py (hunk 3 above)
Comment (sschipke) [reply app.py]: Is it ok if we keep them? I think they are handy, and we could definitely use them in the future.

File: .travis.yml (hunk 2 above)
Comment (nancylee713) [reply .travis.yml]: just the word `deploy` (since it is redundant)

File: app.py (hunk 3 above)
Comment (nancylee713) [reply app.py]: MHM!

File: .travis.yml (hunk 2 above)
Comment (sschipke) [reply .travis.yml]: Done. Thank you!

File: app.py (hunk 3 above)
Comment (sschipke) [reply app.py]: Thanks. If we work on our future iterations, we will definitely need these.

File: tests/test_endpoints.py [hunk 5]
```diff
@@ -63,11 +64,11 @@ def test_get_grammar_questions_by_level(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(len(data), 5)
 
-    def sad_test_get_grammar_questions_by_wrong_level(self):
+    def test_sad_get_grammar_questions_by_wrong_level(self):
         response = self.app.get(BAD_GRAMMAR_QUESTIONS_BY_LEVEL_URL)
         data = json.loads(response.get_data())
         self.assertEqual(response.status_code, 404)
-        self.assertEqual({"error":"Cannot get "})
+        self.assertEqual(data, {'error': 'Level must be either beginner, intermediate, or advanced NOT -hard-'})
```
Comment (nancylee713) [inline tests/test_endpoints.py]: We may want to come back to error handling in Flask later

**Pull request merged**